### ISSUE TITLE
zig cc fails with upx-test-build-with-zig

### ISSUE THREAD

**Issue by markus-oberhumer** [ziglang/zig#13250]
Please have a look at  https://github.com/upx/upx-test-build-with-zig, a tiny repository that contains one GitHub Actions file to build the C++ app [UPX](https://github.com/upx/upx) for 51 different targets, and also tests a number of them using `qemu`.

And see https://github.com/upx/upx-test-build-with-zig/actions/runs/3299582346 for actual results and logs.

Ever since I stumbled across https://andrewkelley.me/post/zig-cc-powerful-drop-in-replacement-gcc-clang.html I had thought about trying `zig cc`  in practice, and now that a new UPX release is imminent and llvm-15 has landed in Zig I had spent the last days to get this working.

I will have a closer look at the actual logs and try to dissect some actually useful bug reports over the next time - the problems seem to range from just unsupported/wrong targets to libstdc++ issues and even possible codegen or qemu bugs (`i386-linux-musl`).

Feedback is highly welcome!

Cheers,
Markus

P.S. Sorry for the nonsensical bug report title

**Comment by markus-oberhumer** [comment]
`arm-linux-musleabi` and `arm-linux-musleabihf`:

these seem to be blocked by https://github.com/ziglang/zig/issues/6573

I have not found a `zig c++` compiler option to disable multithreading, tough.

**Comment by markus-oberhumer** [comment]
`armeb-linux-musleabi` and  `armeb-linux-musleabihf`:

 ld.lld: error: unknown emulation: armebelf_linux_eabi

**Comment by markus-oberhumer** [comment]
`i386-linux-musl`:

now this is strange - the Debug build fails under qemu. Have to check on real hardware

**Comment by markus-oberhumer** [comment]
`mips64-linux-musl`:

/home/runner/.local/bin/zig-linux-x86_64-0.10.0-dev.4476+0f0076666/lib/std/os/linux.zig:50:34: error: struct 'os.linux.arch_bits__struct_4871' has no member named 'syscall3'

**Comment by markus-oberhumer** [comment]
`powerpc-linux-musl`,  `powerpc64-linux-musl` and `powerpc64le-linux-musl`:

zig: error: argument unused during compilation: '-mred-zone' [-Werror,-Wunused-command-line-argument]

This is fatal because we compile with `-Werrorr`.

**Comment by markus-oberhumer** [comment]
`s390x-linux-musl`:

error: unknown target CPU 'generic'
note: valid target CPU values are: arch8, z10, arch9, z196, arch10, zEC12, arch11, z13, arch12, z14, arch13, z15, arch14, z16

**Comment by markus-oberhumer** [comment]
`aarch64-windows-gnu`:

Warnings in mingw secapi. Errors in libunwind.

**Comment by markus-oberhumer** [comment]
`arm-linux-gnueabi` and `arm-lunux-gnueabihf`:

/home/runner/.local/bin/zig-linux-x86_64-0.10.0-dev.4476+0f0076666/lib/libc/glibc/sysdeps/arm/sysdep.h:25:11: fatal error: 'arm-features.h' file not found
`# include <arm-features.h>`

**Comment by markus-oberhumer** [comment]
`armeb-linux-gnueabi` and `armeb-linux-gnueabihf`:

ld.lld: error: unknown emulation: armebelf_linux_eabi 

(same error as musl above)

**Comment by markus-oberhumer** [comment]
`i386-linux-gnu.2.3.4`:

ld.lld: error: undefined symbol: __stack_chk_fail
ld.lld: error: undefined symbol: __isoc99_sscanf

**Comment by markus-oberhumer** [comment]
`i386-windows-gnu`: 

lld-link: error: <root>: undefined symbol: _tls_index

**Comment by markus-oberhumer** [comment]
(More to come :smile: )

**Comment by andrewrk** [comment]
Thanks for the report!

**Comment by nektro** [comment]
> the problems seem to range from just unsupported/wrong targets [...]

note that zig cc CLI expects `-target` not `--target` like you have in your repo

**Comment by andrewrk** [comment]
> note that zig cc CLI expects `-target` not `--target` like you have in your repo

Incorrect. The `--target=foo` syntax is correctly understood by zig cc.
